src/Components/AddItem.jsx

```diff
@@ -1,26 +1,63 @@
 import React from 'react';
-import { useForm } from 'react-hook-form';
-import firebase from 'firebase';
 import { db } from '../lib/firebase';
+import { useForm } from 'react-hook-form';
+import { formatString } from '../lib/helpers.js';
 
 export default function AddItem({ token }) {
-  const { register, handleSubmit, reset, errors } = useForm();
-  const onSubmit = (data) => {
-    const currentList = db.collection('lists').doc(token);
+  const { register, handleSubmit, reset, setError, errors } = useForm();
+
+  const checkForDuplicateItem = async (currentList, itemName) => {
+    const currentListData = await currentList.get();
+    const itemList = currentListData.data();
+    if (itemList[itemName]) {
+      return true;
+    }
+    return false;
+  };
+
+  const updateFirestore = (currentList, itemName, formData) => {
     currentList
       .update({
-        items: firebase.firestore.FieldValue.arrayUnion({
-          name: data.itemName,
-          frequency: parseInt(data.itemFrequency),
+        [itemName]: {
```

Comment:
I'm curious about the refactoring that you've done here 🤔 It looks like this change might have caused the items to be saved differently than on `main`, where they save to an `items` array - on this branch, each item saves as a separate object. 

For example, see the list for token `beth agree eclat`:
<img width="265" alt="Screen Shot 2020-10-22 at 9 10 23 PM" src="https://user-images.githubusercontent.com/48691862/96955971-ca3eab80-14ab-11eb-8f7a-0c99d6b53380.png">

I added coffee on `main`, where it saved to `items`, and I added decaf coffee on this branch, where it saved as a separate `decafcoffee` object.

The items also display in the list page differently, because `main` looks for `items, so it displays this:
<img width="106" alt="Screen Shot 2020-10-22 at 9 12 12 PM" src="https://user-images.githubusercontent.com/48691862/96956112-243f7100-14ac-11eb-9a05-248b3b1a4986.png">

The feature branch is looking for top-level objects, so it displays this:
<img width="144" alt="Screen Shot 2020-10-22 at 9 10 39 PM" src="https://user-images.githubusercontent.com/48691862/96956099-1be73600-14ac-11eb-9502-7c7c72baabe0.png">

It looks like this was intended from your PR description, so I wanted to double-check that it's part of the plan for merging the two branches?

Reply:
We switched from storing the items in an array to using an object to make it a little easier to search for item names. Once we do our merging this week, we'll definitely clean up the database so that each list stores items the same way.

Reply:
Ahh, yep, that makes sense, thank you 👍 That's actually a good example of something to put in the test/PR notes, for both reviewers/testers, and your future selves when you're merging!